app/javascript/components/WorkflowsPage/WorkflowList.tsx

```diff
@@ -65,18 +63,15 @@ const WorkflowList = () => {
                   ) : (
                     <Button
                       color="danger"
-                      onClick={asyncVoid(async () => {
-                        try {
-                          setDeletingWorkflow({ ...deletingWorkflow, state: "deleting" });
-                          await deleteWorkflow(deletingWorkflow.id);
-                          setWorkflows(workflows.filter((workflow) => workflow.external_id !== deletingWorkflow.id));
-                          setDeletingWorkflow(null);
-                          showAlert("Workflow deleted!", "success");
-                        } catch (e) {
-                          assertResponseError(e);
-                          showAlert("Sorry, something went wrong. Please try again.", "error");
-                        }
-                      })}
+                      onClick={() => {
+                        setDeletingWorkflow({ ...deletingWorkflow, state: "deleting" });
+                        router.delete(Routes.workflow_path(deletingWorkflow.id), {
+                          only: ["workflows", "flash"],
+                          onFinish: () => {
+                            setDeletingWorkflow(null);
+                          },
```

Comment:
If the request fails, we need to display the "Something went wrong message" like in the [old behavior](https://github.com/antiwork/gumroad/pull/1902/files#diff-e470a3efc59cd97c0aab2fcc4f190f62e227351f25533fa7b9c601a732a2b048L77).